network/message/authorization.go

```diff
@@ -0,0 +1,364 @@
+// message auth Config.
+// Expected error returns during normal operations:
+//  * ErrUnknownMsgType : if underlying type of v does  not match any of the known message types
+func GetMessageAuthConfig(v interface{}) (MsgAuthConfig, error) {
+	switch v.(type) {
+	// consensus
+	case *messages.BlockProposal:
+		return authorizationConfigs[BlockProposal], nil
+	case *messages.BlockVote:
+		return authorizationConfigs[BlockVote], nil
+
+	// protocol state sync
+	case *messages.SyncRequest:
+		return authorizationConfigs[SyncRequest], nil
+	case *messages.SyncResponse:
+		return authorizationConfigs[SyncResponse], nil
+	case *messages.RangeRequest:
+		return authorizationConfigs[RangeRequest], nil
+	case *messages.BatchRequest:
+		return authorizationConfigs[BatchRequest], nil
+	case *messages.BlockResponse:
+		return authorizationConfigs[BlockResponse], nil
+
+	// cluster consensus
+	case *messages.ClusterBlockProposal:
+		return authorizationConfigs[ClusterBlockProposal], nil
+	case *messages.ClusterBlockVote:
+		return authorizationConfigs[ClusterBlockVote], nil
+	case *messages.ClusterBlockResponse:
+		return authorizationConfigs[ClusterBlockResponse], nil
+
+	// collections, guarantees & transactions
+	case *flow.CollectionGuarantee:
+		return authorizationConfigs[CollectionGuarantee], nil
+	case *flow.TransactionBody:
+		return authorizationConfigs[TransactionBody], nil
+	case *flow.Transaction:
+		return authorizationConfigs[Transaction], nil
+
+	// core messages for execution & verification
+	case *flow.ExecutionReceipt:
+		return authorizationConfigs[ExecutionReceipt], nil
+	case *flow.ResultApproval:
+		return authorizationConfigs[ResultApproval], nil
+
+	// execution state synchronization
+	case *messages.ExecutionStateSyncRequest:
+		return authorizationConfigs[ExecutionStateSyncRequest], nil
+	case *messages.ExecutionStateDelta:
+		return authorizationConfigs[ExecutionStateDelta], nil
+
+	// data exchange for execution of blocks
+	case *messages.ChunkDataRequest:
+		return authorizationConfigs[ChunkDataRequest], nil
+	case *messages.ChunkDataResponse:
+		return authorizationConfigs[ChunkDataResponse], nil
+
+	// result approvals
+	case *messages.ApprovalRequest:
+		return authorizationConfigs[ApprovalRequest], nil
+	case *messages.ApprovalResponse:
+		return authorizationConfigs[ApprovalResponse], nil
+
+	// generic entity exchange engines
+	case *messages.EntityRequest:
+		return authorizationConfigs[EntityRequest], nil
+	case *messages.EntityResponse:
+		return authorizationConfigs[EntityResponse], nil
+
+	// testing
+	case *message.TestMessage:
+		return authorizationConfigs[TestMessage], nil
+
+	// dkg
+	case *messages.DKGMessage:
+		return authorizationConfigs[DKGMessage], nil
+
+	default:
+		return MsgAuthConfig{}, NewUnknownMsgTypeErr(v)
+	}
+}
+
+// GetAllMessageAuthConfigs returns all the configured message auth configurations.
+func GetAllMessageAuthConfigs() []MsgAuthConfig {
+	configs := make([]MsgAuthConfig, 0)
```

Comment:
specifying the known length will avoid unnecessary allocation and GC overhead:
```suggestion
	configs := make([]MsgAuthConfig, 0)	configs := make([]MsgAuthConfig, 0, len(authorizationConfigs))
```